Makefile

```diff
@@ -20,35 +19,61 @@ $(CMD): $(GO_SOURCES)
 ### PACKAGE / RELEASE ###
 #########################
 
-PACKAGES = dist/linux_amd64 dist/linux_arm64
+# Which operating systems and architectures will we cross-compile for?
+# In the form of <os>-<arch>
+OSARCHS=\
+	linux-amd64   \
+	linux-arm64   \
 
-# package: each target in $PACKGAES built into a directory under dist/
-.PHONY: package
-package: $(PACKAGES)
-$(PACKAGES): dist/% : dist/%/$(CMD) dist/%/$(CMD).sha256
+artifacts:
+	@mkdir -p artifacts
+
+# each item in OSARCHS appended to "artifacts/influx2hny-" to
+# generate the list of file paths for binaries to be cross-compiled
+BINARIES=$(OSARCHS:%=artifacts/$(CMD)-%)
+
+# these targets set OS and ARCH variables for any target that matches a path in $(BINARIES)
+# ex: "artifacts/influx2hny-linux-arm64"
+# 0: "artifacts/influx2hny"
+# 1: "linux"
+# 2: "arm64"
+# $* is the matched pattern, which is split on '-' to get the OS & ARCH values
+$(BINARIES): OS = $(word 1,$(subst -, ,$*))
+$(BINARIES): ARCH = $(word 2,$(subst -, ,$*))
+
+# builds the actual binary for each OS/ARCH listed in $OSARCHS
+$(BINARIES): artifacts/$(CMD)-%: artifacts
+	@echo "building: $@"
+	GOOS=$(OS) GOARCH=$(ARCH) \
+	  go build -trimpath \
+	           -ldflags "-s -w -X main.BuildVersion=$(RELEASE_VERSION)" \
+	           -o $@ \
+	           ./cmd/$(CMD)
```

Comment:
I feel dirty mixing tabs (because Makefiles) and spaces (because not every editor has its tab-width set the same), but doesn't breaking this command over multiple lines with indentation make it easier to read?